### DISCUSSION TITLE
Running SRW on Cheyenne

### DISCUSSION THREAD

**Discussion by htan2013** [ufs-community/ufs-srweather-app#999] [General]
Dear all experts,

I am wondering where is the pre-built SRW in Cheyenne. The guide shows that it is in /glade/scratch/epicufsrt/containers but there is only 2.1.0 there (50MB). Another question is how to load the SRW 2.2.0 environment in Cheyenne. For the previous version, I used the following command to load the environment: 

![image](https://github.com/ufs-community/ufs-srweather-app/assets/77926698/e62c519b-408b-44bf-942a-8c5609770ab1)

Does this work for 2.2.0?

Thanks in advance,
Tan

**Comment by gspetro-NOAA** [comment]
Hi Tan,

Thanks for bringing this to our attention! My understanding is that Cheyenne has been decommissioned and replaced by Derecho. Even if you are able to access Cheyenne, CISL and EPIC support for running on Cheyenne ended on Dec. 31, 2023, so we highly recommend moving to Derecho. The correct container location on Derecho is `/glade/work/epicufsrt/contrib/containers`. (Since Cheyenne and Derecho share a file system, that container path should work on Cheyenne, too.) However, there is a problem with the container on Derecho/Cheyenne, so it is best to use the SRW App "on-prem" rather than in a container on those systems. It also looks like this is what you did in the past--used someone else's on-prem clone of the SRW App, not a container. 

There are no standard locations for a pre-built on-prem SRW App, but it is relatively simple to build on a Level 1 system such as Derecho. Since Derecho already has the prerequisite software installed, you could follow the instructions in the [Quick Start Guide](https://ufs-srweather-app.readthedocs.io/en/release-public-v2.2.0/BuildingRunningTesting/Quickstart.html#building-and-running-the-ufs-srw-application) to build and run the SRW App in your own project directory. The build takes some time, but since it's your own build, you have control over it and won't need to worry about whether someone else changes or deletes it. It's also just a few commands to build (you can do something else while the build runs in the background): 
```
git clone -b release/public-v2.2.0 https://github.com/ufs-community/ufs-srweather-app.git
cd ufs-srweather-app
./manage_externals/checkout_externals
./devbuild.sh --platform=derecho
```

If you choose to build the SRW App v2.2.0 this way, the steps to load the environment are: 
```
module use /path/to/ufs-srweather-app/modulefiles
module load wflow_derecho
conda activate workflow_tools
```
Make sure to replace `/path/to/ufs-srweather-app/modulefiles` with the actual path (e.g., `/glade/work/projectname/htan/ufs-srweather-app/modulefiles` or something similar).

**Reply by htan2013** [reply]
Hi Gillian,

Thank you so much for your explanation! I know that they will migrate everything from Cheyenne to Derecho so I was wondering whether we have a container. 

Best regards,
Haochen
